app/assets/javascripts/app/views.js

```diff
@@ -103,28 +103,27 @@ app.views.Base = Backbone.View.extend({
   },
 
   report: function(evt) {
-    if(evt) { evt.preventDefault(); }
-    var msg = prompt(Diaspora.I18n.t('report.prompt'), Diaspora.I18n.t('report.prompt_default'));
-    if (msg == null) {
-      return;
-    }
-    var data = {
-      report: {
-        item_id: this.model.id,
-        item_type: $(evt.currentTarget).data("type"),
-        text: msg
-      }
+    if (evt) { evt.preventDefault(); }
+    var report = {
+      item_id: this.model.id,
+      item_type: $(evt.currentTarget).data("type"),
+      text: "Not filled"
     };
+    document.getElementById("report-content-form").addEventListener("submit", function(ev) {
```

Comment:
We should make sure to disallow sending reports without a reason submitted. I see a lot of reports using the default value in my database, and those reports are more or less useless. Forcing people to at least come up with an explanation is a good thing.

Reply:
The input has the `required` attribute, so any modern browser won't allow the user to submit the form if no reason is given. And it is empty by default. Do you think it is still worth doing a check in Javascript? Or even server side?

Reply:
> Or even server side?

*Always* validate state on the server side.

Reply:
What I meant was, in that case, I didn't change anything on the server side. Anyway I just tried and it already fails if the report content is empty. So is server side handling + `required` in the markup enough, or do we want to double check on the client side, for very old browsers?

Reply:
That sounds like a good-enough implementation, then. :)